skbio/metadata/_interval.py

```diff
@@ -530,38 +569,29 @@ def num_interval_features(self):
         '''The total number of interval features.'''
         return len(self._intervals)
 
-    def _rebuild_tree(method):
-        """Rebuild the IntervalTree."""
-        @functools.wraps(method)
-        def inner(self, *args, **kwargs):
-            if self._is_stale_tree is False:
-                return method(self, *args, **kwargs)
-            self._interval_tree = IntervalTree()
-            for f in self._intervals:
-                for start, end in f.bounds:
-                    self._interval_tree.add(start, end, f)
-            self._is_stale_tree = False
-            return method(self, *args, **kwargs)
-        return inner
-
     def _reverse(self):
         """Reverse ``IntervalMetadata`` object.
 
         This operation reverses all of the interval coordinates.
         For instance, this can be used to compare coordinates
         in the forward strand to coordinates in the reversal strand.
         """
+
         for f in self._intervals:
-            intvls = [(self.upper_bound - x[1], self.upper_bound - x[0])
-                      for x in reversed(f.bounds)]
+            try:
+                intvls = [(self.upper_bound - x[1], self.upper_bound - x[0])
+                          for x in reversed(f.bounds)]
+            except TypeError:
```

Comment:
I feel like this is more pythonic because of [EAFP](https://docs.python.org/3/glossary.html#term-eafp)?